mesonbuild/compilers/cuda.py

```diff
@@ -60,23 +61,423 @@ def __init__(self, exelist: T.List[str], version: str, for_machine: MachineChoic
         self.warn_args = {level: self._to_host_flags(flags) for level, flags in host_compiler.warn_args.items()}
 
     @classmethod
-    def _to_host_flags(cls, flags: T.List[str], phase: _Phase = _Phase.COMPILER) -> T.List[str]:
-        return [cls._to_host_flag(f, phase=phase) for f in flags]
+    def _shield_nvcc_list_arg(cls, arg: str, listmode: bool=True) -> str:
+        """
+        Shield an argument against both splitting by NVCC's list-argument
+        parse logic, and interpretation by any shell.
+
+        NVCC seems to consider every comma , that is neither escaped by \ nor inside
+        a double-quoted string a split-point. Single-quotes do not provide protection
+        against splitting; In fact, after splitting they are \-escaped. Unfortunately,
+        double-quotes don't protect against shell expansion. What follows is a
+        complex dance to accomodate everybody.
+        """
+
+        SQ = "'"
+        DQ = '"'
+        CM = ","
+        BS = "\\"
+        DQSQ = DQ+SQ+DQ
+        quotable = set(string.whitespace+'"$`\\')
+
+        if CM not in arg or not listmode:
+            if SQ not in arg:
+                """
+                If any of the special characters "$`\ or whitespace are present, single-quote.
+                Otherwise return bare.
+                """
```

Comment:
These aren't docstrings, they should use the `# comment format`. A docstring must come directly after a `def` or `class` definition

Reply:
> These aren't docstrings, they should use the `# comment format`. A docstring must come directly after a `def` or `class` definition

This is true, but triple-quoted strings elsewhere are simply assignments of a string literal to nothing in particular.

I can certainly convert them all to packs of `#`-prefixed comments.